app/src/main/java/org/oppia/android/app/activity/route/ActivityRouterModule.kt

```diff
@@ -0,0 +1,19 @@
+package org.oppia.android.app.activity.route
+
+import dagger.Module
+import dagger.Provides
+import dagger.multibindings.IntoMap
+import dagger.multibindings.StringKey
+import org.oppia.android.app.model.DestinationScreen
+
+/** RouterModule to bind [DestinationScreen]. */
+@Module
+class ActivityRouterModule {
```

Comment:
Is this module even being used? I don't see the injection for the map anywhere in the PR. Am I missing something?

Reply:
> @rt4914 it seems like you're putting an extra bundle in the verify part (the thing associated with the key isn't a bundle, it's a proto `ByteString`). I think what we need here is our own version of hasExtra() that matches against the proto itself, and probably in a type-safe way. One option (unverified code):
> 
> ```kotlin
> private fun <T: MessageLite> hasProtoExtra(keyName: String, expectedProto: T): Matcher<Intent> {
>   return hasExtra(keyName, expectedProto.toByteString())
> }
> ```

Still I tests are failing. I have tested it.

Reply:
Still unresolved I think.

Reply:
Updated this based on the the code @BenHenning provided